src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

```diff
@@ -551,3 +551,53 @@ services:
         tags:
             - { name: form.type }
 
+    form.type.catalog.abstract_category:
+        class: 'PrestaShopBundle\Form\Admin\Catalog\Category\AbstractCategoryType'
+        abstract: true
+        parent: 'form.type.translatable.aware'
+        arguments:
+            - '@=service("prestashop.core.form.choice_provider.group_by_id").getChoices()'
+            - '@prestashop.adapter.multistore_feature'
+        public: true
+
+    form.type.catalog.category:
+        class: 'PrestaShopBundle\Form\Admin\Catalog\Category\CategoryType'
+        parent: 'form.type.catalog.abstract_category'
+        public: true
+        tags:
+          - { name: form.type }
+
+    form.type.catalog.root_category:
+        class: 'PrestaShopBundle\Form\Admin\Catalog\Category\RootCategoryType'
+        parent: 'form.type.catalog.abstract_category'
+        public: true
+        tags:
+            - { name: form.type }
+
+    form.type.common.translate_textarea:
+        class: 'PrestaShopBundle\Form\Admin\Type\TranslateTextareaType'
+        parent: 'form.type.translatable.aware'
+        public: true
+        tags:
+            - { name: form.type }
+
+    form.type.common.category_choice_tree:
+        class: 'PrestaShopBundle\Form\Admin\Type\CategoryChoiceTreeType'
+        arguments:
+            - '@prestashop.core.api.category.repository'
+        tags:
+            - { name: form.type }
+
+    form.type.common.translatable:
+        class: 'PrestaShopBundle\Form\Admin\Type\TranslatableType'
+        arguments:
+            - "@=service('prestashop.adapter.legacy.context').getLanguages()"
+        tags:
+            - { name: form.type }
+
+    form.type.common.shop_choice_tree:
+        class: 'PrestaShopBundle\Form\Admin\Type\ShopChoiceTreeType'
+        arguments:
+            - "@=service('prestashop.adapter.form.choice_provider.shop_tree_choice_provider').getChoices()"
+        tags:
+            - { name: form.type }
```

Comment:
This file becomes and will be bigger and bigger, should we split this one?

Reply:
Yes ^^ 600 lines is enough indeed

Reply:
:+1: but not in this PR ^^